mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala

```diff
@@ -237,4 +253,61 @@ object Tool extends App with LazyLogging {
         Tool.ERROR_EXIT_CODE
     }
   }
+
+  private def printHeaderAndConfig(): Unit = {
+    Console.println("# APALACHE version %s build %s".format(Version.version, Version.build))
+    Console.println("#")
+    Console.println("# WARNING: This tool is in the experimental stage.")
+    Console.println("#          Please report bugs at: " + ISSUES_LINK)
+    Console.println("# ")
+
+    if (ExecutionStatisticsCollector.promptUser()) {
+      // Statistics collection is not enabled. Cry for help.
+      Console.println("# Usage statistics is OFF. We care about your privacy.")
+      Console.println("# If you want to help our project, consider enabling statistics with config --submit-stats=true.")
+    } else {
+      // Statistic collection is enabled. Thank the user
+      Console.println("# Usage statistics is ON. Thank you!")
+      Console.println("# If you have changed your mind, disable the statistics with config --submit-stats=false.")
+    }
+    Console.println("")
+  }
+
+  private def configure(config: ConfigCmd): Unit = {
+    config.submitStats match {
+      case Some(isEnabled) =>
+        val statCollector = new ExecutionStatisticsCollector()
+        if (isEnabled) {
+          logger.info("Statistics collection is ON.")
+          logger.info("This also enabled TLC and TLA+ Toolbox statistics.")
+          statCollector.set(Selection.ON)
+        } else {
+          logger.info("Statistics collection is OFF.")
+          logger.info("This also disabled TLC and TLA+ Toolbox statistics.")
+          statCollector.set(Selection.NO_ESC)
+        }
+
+      case None =>
+        ()
+    }
+  }
+
+  // We are collecting statistics with the code from tlatools:
+  //
+  // https://github.com/tlaplus/tlaplus/blob/master/tlatools/org.lamport.tlatools/src/util/ExecutionStatisticsCollector.java
+  private def submitStatisticsIfEnabled(commandParams: Map[String, String]): Unit = {
+    val statCollector = new ExecutionStatisticsCollector()
+    if (statCollector.isEnabled) {
+      val params = new java.util.HashMap[String, String]()
+      params.put("ver", "apalache-%s-%s".format(Version.version, Version.build))
```

Comment:
I could also add an (optional) field to the stats backend that discriminates Apalache and TLC (similar to the IDE flag).

Reply:
recommendations for the field name?

Reply:
If you don't want to report `mode` (TLC: bfs/dfs/sim), you could hijack this field.

Reply:
For apalache, mode will be `apalache/parse`, `apalache/check`, and `apalache/typecheck`.

Reply:
I'm in the process of adding a tool filed/column so you can pass "apalache".  I'd suggest to report `mode` as "check", "parse", or ''typecheck".  Some part will probably choke on the '/' otherwise.

Reply:
The backend now has a 'tool' field.

Reply:
Cool. Now Apalache is reporting both `tool` and `mode`